lib/consolidate.js

```diff
@@ -91,13 +93,24 @@ function read(path, options, fn) {
   if (cached) return fn(null, str);
 
   // read
-  fs.readFile(path, 'utf8', function(err, str){
-    if (err) return fn(err);
-    // remove extraneous utf8 BOM marker
-    str = str.replace(/^\uFEFF/, '');
-    if (options.cache) readCache[path] = str;
-    fn(null, str);
-  });
+  if (nowReading[path]) {
+    if (!readListeners[path]) readListeners[path] = [fn];
+    else readListeners[path].push(fn);
+  }else{
+    nowReading[path] = true;
+    fs.readFile(path, 'utf8', function(err, str){
+      if (err) return fn(err);
```

Comment:
@fantapop I created a module based on @mhiguera original code in this PR and submitted a new PR with the update (so it gets all the new code since this PR).

Will you try this in your metalsmith-templates project (might have to remove their version of consolidate to use this branch) and let me know if it works for you?